shell/platform/windows/text_input_plugin_unittest.cc

```diff
@@ -366,6 +368,60 @@ TEST(TextInputPluginTest, TextEditingWorksWithDeltaModel) {
   // Passes if it did not crash
 }
 
+TEST(TextInputPluginTest, CompositionCursorPos) {
+  int selection_base = -1;
+  TestBinaryMessenger messenger([&](const std::string& channel,
+                                    const uint8_t* message, size_t size,
+                                    BinaryReply reply) {
+    auto method = JsonMethodCodec::GetInstance().DecodeMethodCall(
+        std::vector<uint8_t>(message, message + size));
+    if (method->method_name() == kUpdateEditingStateMethod) {
+      const auto& args = *method->arguments();
+      const auto& editing_state = args[1];
+      auto base = editing_state.FindMember(kSelectionBaseKey);
+      ASSERT_NE(base, editing_state.MemberEnd());
+      ASSERT_TRUE(base->value.IsInt());
+      selection_base = base->value.GetInt();
+    }
+  });
+  MockTextInputPluginDelegate delegate;
+
+  TextInputPlugin plugin(&messenger, &delegate);
+
+  auto args = std::make_unique<rapidjson::Document>(rapidjson::kArrayType);
+  auto& allocator = args->GetAllocator();
+  args->PushBack(123, allocator);  // client_id
+  rapidjson::Value client_config(rapidjson::kObjectType);
+  args->PushBack(client_config, allocator);
+  auto encoded = JsonMethodCodec::GetInstance().EncodeMethodCall(
+      MethodCall<rapidjson::Document>(kSetClientMethod, std::move(args)));
+  EXPECT_TRUE(messenger.SimulateEngineMessage(
+      kChannelName, encoded->data(), encoded->size(),
+      [](const uint8_t* reply, size_t reply_size) {}));
+
+  plugin.ComposeBeginHook();
+  EXPECT_EQ(selection_base, 0);
```

Comment:
nit: maybe we can also check `selection_extent` here.

Reply:
In all test cases, the selection should be collapsed, so I will expect that `extent == base` in the messenger callback.